torchdata/nodes/loader.py

```diff
@@ -10,10 +10,23 @@ def __init__(self, root: BaseNode[T], restart_on_stop_iteration: bool = True):
         self.restart_on_stop_iteration = restart_on_stop_iteration
         self._next_iter_state_dict: Optional[Dict[str, Any]] = None
         self._it: Optional[LoaderIterator[T]] = None
+        # Tracks whether an iterator was created solely for getting a state_dict, in which case
```

Comment:
Can confirm that with this change to the loader.py we do not need to tweak the test here https://github.com/pytorch/data/pull/1369/files#diff-b4616338bda203d1cf7ac17ae95afaa662502e89f5b8b9bb6c3e75e7bb6ad73aR125-R127 :-)